src/catalog/catalog.cpp

```diff
@@ -45,6 +56,27 @@ void Catalog::DeleteDatabase(transaction::TransactionContext *txn, const std::st
 
   // delete from the map
   cat_map_.erase(oid);
+  return true;
+}
+
+bool Catalog::DeleteDatabase(transaction::TransactionContext *txn, db_oid_t db) {
+  // get database handle
+  auto db_handle = GetDatabaseHandle();
+  auto db_entry = db_handle.GetDatabaseEntry(txn, db);
+  if (db_entry == nullptr) return false;
+
+  // remove entry from pg_database
+  bool ret = db_handle.DeleteEntry(txn, db_entry);
+  if (!ret) return false;
+
+  // TODO(pakhtar): delete all user tables
+  // destroy all the non-global tables
+  // this should become just catalog tables
+  DeleteDatabaseTables(db);
```

Comment:
Once we have foreign keys, this will fail. You will need to delete the tables first, then delete the database.